brainframe/api/bf_codecs/alarm_codecs.py

```diff
@@ -48,24 +48,30 @@ def from_dict(d):
 
 
 class Alert(Codec):
-    """This is sent when an Alarm has been triggered.
-    An alert can be sent WITHOUT an end_time, but never without a start_time.
-
-    self.verified_as can be True, False or None.
-        If True, then this alert was labeled by a person as being legitimate
-        If False,then this alert was labeled by a person as being a false alarm
-        If None, then this alert has not been labeled yet.
-    """
+    """This is sent when an Alarm has been triggered."""
 
     def __init__(self, *, alarm_id, zone_id, stream_id, start_time, end_time,
                  verified_as, id_=None):
-        self.id = id_
-        self.alarm_id = alarm_id
-        self.zone_id = zone_id
-        self.stream_id = stream_id
-        self.start_time = start_time
-        self.end_time = end_time
-        self.verified_as = verified_as
+        self.id: int = id_
```

Comment:
Why type hint the attributes instead of the arguments? Does PyCharm give proper Quick Documentation with Ctrl-Q?

Reply:
> What if we made type annotating the __init__ arguments a best-effort thing, but concentrated on type annotating the fields as the main source of truth? I think the additional type annotation in __init__ provides enough value to at least take a stab at doing it.

Ok so looking into it a bit more, I disagree with this. What we've documented for the most part so far are the _fields_ of a class. These do not necessarily match the arguments used for its __init__ function. I think both should be documented.

Reply:
I think having docstrings for both the `__init__` arguments and the fields isn't really feasible to maintain. I believe that they're the same in all cases for codecs as well, since they're essentially dataclasses.

Reply:
Just to give a little more detail on why I think there would be a maintenance problem: I think the docs for the field and the `__init__` argument would be the same in almost all cases. It would be nice for tooling but it would lead to a lot of duplicate information when reading on ReadTheDocs and would require us to remember to update in both places whenever we make a change to those docstrings.

Reply:
Maybe we should just be using the [dataclasses backport](https://pypi.org/project/dataclasses/) 🤔

Reply:
Oh, I like that quite a bit. I'll try it out!